fcrepo4-exts/fcrepo-upgrade-utils
# Initial work on 4.7.x to 5.x upgrade.

Requires Java 11.

It supports upgrading containers, binaries and external content.
Currently, versions and acls are not upgraded properly.
Direct and indirect containers have not been tested.

Resolves: https://jira.duraspace.org/browse/FCREPO-3036

**Initial work on 4.7.x to 5.x upgrade.**
* * *

**JIRA Ticket**: https://jira.duraspace.org/browse/FCREPO-3036


* Other Relevant Links (Mailing list discussion, related pull requests, etc.)

# What does this Pull Request do?

It supports upgrading containers, binaries and external content.
Currently, versions and acls are not upgraded properly.
Direct and indirect containers have not been tested.

# How should this be tested?
There are some unit tests that run the upgrade on a minimal 4.7.5  export.
To test manually, you'll need to:

1. Upgrade the sample data:
`java -Dfcrepo.log=DEBUG -jar target/fcrepo-upgrade-utils-5.0.0-SNAPSHOT.jar -i src/test/resources/4.7.5-export`
2.  fire up clean 5.1.0 instance.
3. You'll need to overwrite the root acl due to a minor bug in fcrepo4: 
create a file named `read-only-root.ttl` with the following contents:
```
@prefix acl: <http://www.w3.org/ns/auth/acl#> .

<#authorization>    a               acl:Authorization;
    acl:accessTo    </rest/>;
    acl:default </rest/>;
    acl:mode       acl:Read, acl:Write;
    acl:agent       "fedoraAdmin".
```
then PUT it into place: 
curl -u fedoraAdmin:fedoraAdmin http://localhost:8080/rest/fcr:acl -d "@read-only-root.ttl" -X PUT -H "Content-Type: text/turtle"

4. use the[ fcrepo-import-export tool v.0.3.0](https://github.com/fcrepo4-labs/fcrepo-import-export/releases/tag/fcrepo-import-export-0.3.0) to **import** the resources in  `src/test/resources/4.7.5-export` into your 5.1.0 instance.
5. Verify the resources in http://localhost:8080/rest
6. Repeat with your own dataset.

# Interested parties
@fcrepo4/committers
